client/trino-jdbc/src/test/java/io/airlift/units/ThreadCount.java

```diff
@@ -0,0 +1,187 @@
+            return new ThreadCount(round(threadCount));
+        }
+
+        return new ThreadCount(parseInteger(value));
+    }
+
+    public static ThreadCount boundedValueOf(String value, String minValue, String maxValue)
+    {
+        ThreadCount parsed = ThreadCount.valueOf(value);
+        ThreadCount min = ThreadCount.valueOf(minValue);
+        ThreadCount max = ThreadCount.valueOf(maxValue);
+
+        if (parsed.compareTo(min) < 0) {
+            return min;
+        }
+
+        if (parsed.compareTo(max) > 0) {
+            return max;
+        }
+        return parsed;
+    }
+
+    private static float parseFloat(String value)
+    {
+        try {
+            return Float.parseFloat(value);
+        }
+        catch (NumberFormatException e) {
+            throw new IllegalArgumentException(format("Cannot parse value '%s' as float", value), e);
+        }
+    }
+
+    private static int parseInteger(String value)
+    {
+        try {
+            long parsed = Long.parseLong(value);
+            checkArgument(parsed <= Integer.MAX_VALUE, "Thread count is greater than 2^32 - 1");
+            return toIntExact(parsed);
+        }
+        catch (NumberFormatException e) {
+            throw new IllegalArgumentException(format("Cannot parse value '%s' as integer", value), e);
+        }
+    }
+
+    @Override
+    public int compareTo(ThreadCount o)
+    {
+        return Integer.compare(threadCount, o.threadCount);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ThreadCount that = (ThreadCount) o;
+        return threadCount == that.threadCount;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return threadCount;
+    }
+
+    @Override
+    public String toString()
+    {
+        return (threadCount == 1) ? "1 thread" : (threadCount + " threads");
+    }
+
+    static final class MachineInfo
+    {
+        private static final Path CPU_INFO_PATH = Paths.get("/proc/cpuinfo");
+
+        // cache physical processor count, so that it's not queried multiple times during tests
+        private static volatile int physicalProcessorCount = -1;
+
+        private MachineInfo() {}
+
+        public static int getAvailablePhysicalProcessorCount()
+        {
+            if (physicalProcessorCount != -1) {
+                return physicalProcessorCount;
+            }
+
+            String osArch = System.getProperty("os.arch");
+            // logical core count (including container cpu quota if there is any)
+            int availableProcessorCount = Runtime.getRuntime().availableProcessors();
+            int totalPhysicalProcessorCount = availableProcessorCount;
+            if ("amd64".equals(osArch) || "x86_64".equals(osArch)) {
+                OptionalInt procInfo = tryReadFromProcCpuinfo();
```

Comment:
How is it better than `Runtime.getRuntime().availableProcessors()` and why take `min` of the two instead of just using "the better one"? 
This should be documented.